metadata-ingestion/src/datahub/ingestion/source/unity/source.py

```diff
@@ -172,7 +172,18 @@ def get_workunits_internal(self) -> Iterable[MetadataWorkUnit]:
         yield from self.process_metastores()
 
     def process_metastores(self) -> Iterable[MetadataWorkUnit]:
+        metastores: List[Metastore] = []
+        assigned_metastore = self.unity_catalog_api_proxy.assigned_metastore()
+        if assigned_metastore:
+            metastores.append(assigned_metastore)
         for metastore in self.unity_catalog_api_proxy.metastores():
+            if (
+                assigned_metastore
+                and metastore.metastore_id != assigned_metastore.metastore_id
+            ):
+                metastores.append(metastore)
```

Comment:
If `assigned_metastore` is `None` then we'll get no metastores right? This doesn't seem intended.

You can do `not assigned_metastore or metastore.metastore_id != assigned_metastore.metastore_id` but it might be simpler to just stick the metastores in a map indexed by their id to avoid duplicates